File: 2155-find-missing-observations/2155-find-missing-observations.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int sum = 0;
        for (int roll : rolls) {
            sum += roll;
        }

        int m = rolls.length;
        int missingSum = mean * (n + m) - sum;
        if (missingSum < n || missingSum > n * 6) return new int[0];

        int val = missingSum / n;
        int rem = missingSum % n;
        int[] ans = new int[n];
        Arrays.fill(ans, val);

        for (int i = 0; i < rem; i++) {
            ans[i]++;
        }

        return ans;
    }
}

File: 2155-find-missing-observations/2155-find-missing-observations.py
class Solution(object):
    def missingRolls(self, rolls, mean, n):
        sum_rolls = sum(rolls)
        m = len(rolls)
        missing_sum = mean * (n + m) - sum_rolls
        
        if missing_sum < n or missing_sum > n * 6:
            return []
        
        val = missing_sum // n
        rem = missing_sum % n
        ans = [val] * n
        
        for i in range(rem):
            ans[i] += 1
        
        return ans
        